### PR TITLE
Blogpost about sprint 152

### PR DESCRIPTION
**Pull request by rubhanazeem**

### CODE REVIEW

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+category: development
+---
+
+A lot of stuff happens behind the scenes whenever OBS runs SCM/CI workflow. Sometimes, it gets hard to keep track of things like which workflow failed or completed successfully. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.
+
+{% include partials/_series-of-posts-about-scm-integration.md %}
+
+## Notifications about failed SCM/CI workflows
+
+Whenever SCM/CI workflow executes you get a new entry of WorkflowRuns that gives all the details about status, event type, event source etc. But now OBS has the option to send notification if the WorkflowRun fails.
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:13]:
```suggestion
Whenever an SCM/CI workflow is executed, you get a new entry of workflow run that gives all the details about status, event type, event source etc. But now OBS also gives you the possibility of receiving a notification as soon as one workflow run fails.

Give it a try! Subscribe to the new type of (notifications)[https://build.opensuse.org/my/subscriptions] so you don't have to manually check what's going on with your workflows. OBS will let you know as soon as something wrong happens.
```

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+<figure>
+  <img src="/images/posts/workflow-run-notification-1.png" alt="Screenshot of workflow run notification activation" />
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:16]: This is not the correct image to display here. Email should be enabled as well, we introduced it in a later PR.

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+layout: post
+title: Notifications about failed SCM/CI Workflows and more
+category: development
+---
+
+A lot of stuff happens behind the scenes whenever OBS runs SCM/CI workflow. Sometimes, it gets hard to keep track of things like which workflow failed or completed successfully. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.
+
+{% include partials/_series-of-posts-about-scm-integration.md %}
+
+## Notifications about failed SCM/CI workflows
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:11]: Titleized:

```suggestion
## Notifications About Failed SCM/CI Workflows
```

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+<figure>
+  <img src="/images/posts/workflow-run-notification-1.png" alt="Screenshot of workflow run notification activation" />
+</figure>
+
+<figure>
+  <img src="/images/posts/workflow-run-notification-2.png" alt="Screenshot of failed workflow run" />
+</figure>
+
+## Workflow configuration file and SetFlag step artifacts in workflowRun
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:23]:
```suggestion
## Workflow Configuration File and SetFlag Step Artifacts in Workflow Run
```

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Previously, WorkflowRun only showed the path of workflow configuration file and now you can check the whole content of your workflow configuration file in WorkflowRun.
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:25]:
```suggestion
Previously, the workflow run details only included the URL or path to the workflow configuration file. Now it also includes the file content so you can always check what was the configuration at the time the workflow ran.
```

**Review by saraycp — CHANGES_REQUESTED** [review]
I left a few comments but I'd also propose replacing all the `WorkflowRun` with `workflow run` in all their forms (singular, plural, title, etc)

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: Notifications about failed SCM/CI Workflows and more
+category: development
+---
+
+A lot of stuff happens behind the scenes whenever OBS runs SCM/CI workflow. Sometimes, it gets hard to keep track of things like which workflow failed or completed successfully. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.
```
[the remark this answers is not in the document]
Comment (rubhanazeem) [reply _posts/development/2023-10-30-notifications-failed-workflow-runs.md]: I've addressed this. Please check

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md (hunk 2 above)
Comment (rubhanazeem) [reply _posts/development/2023-10-30-notifications-failed-workflow-runs.md:16]: The image has been updated

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Workflow Configuration File and SetFlag Step Artifacts in Workflow Run
+
+Previously, the workflow run details only included the URL or path to the workflow configuration file. Now it also includes the file content so you can always check what was the configuration at the time the workflow ran.
+
+<figure>
+  <img src="/images/posts/workflow-config.png" alt="Screenshot of workflow configuration file" />
+</figure>
+
+In addition to the configuration file WorkflowRun now also stores the artifacts of SetFlag step.
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:31]:
```suggestion
In addition to the configuration file the workflow run now also stores the artifacts of the SetFlag step.
```

**Review by saraycp — CHANGES_REQUESTED** [review]

File: _posts/development/2023-10-30-notifications-failed-workflow-runs.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: Notifications About Failed SCM/CI Workflows and More
+category: development
+---
+
+A lot of things happen behind the scenes whenever OBS runs an SCM/CI workflow. Sometimes, you might want to know which workflow failed or completed successfully and the only way to do this is visiting the workflow runs UI. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.
```
Comment (saraycp) [inline _posts/development/2023-10-30-notifications-failed-workflow-runs.md:7]:
```suggestion
A lot of things happen behind the scenes whenever OBS runs an SCM/CI workflow. Sometimes, you might want to know which workflow failed or completed successfully and the only way to do this is by visiting the workflow runs UI. This time we tried to improve this and other details in the SCM/CI integration.
```

**Review by saraycp — CHANGES_REQUESTED** [review]

**Review by saraycp — APPROVED** [review]

**Pull request merged**